core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -45,9 +45,14 @@ public class PartitionsTable extends BaseMetadataTable {
     this.schema =
         new Schema(
             Types.NestedField.required(1, "partition", Partitioning.partitionType(table)),
-            Types.NestedField.required(2, "record_count", Types.LongType.get()),
-            Types.NestedField.required(3, "file_count", Types.IntegerType.get()),
-            Types.NestedField.required(4, "spec_id", Types.IntegerType.get()));
+            Types.NestedField.required(4, "spec_id", Types.IntegerType.get()),
```

Comment:
I am not sure about the column order, it feels odd to me to have IDs in 1,4,2,3 order. But that's just personal preference.

Reply:
we have discussed this here 
https://github.com/apache/iceberg/pull/6661#discussion_r1087858684

I think @RussellSpitzer and @szehon-ho agree that there is no impact from column re-ordering. 
Spec-id in the middle really looks odd when we add other counters.